### PR TITLE
fix: Allow to get files from git repository if file name starts with dot

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>

### What does this PR do?
Fixes retrieving files from git repository when filename or directory name starts with dots.

### Screenshot/screencast of this PR

Screenshot from 2022-09-07 18-01-29

Screenshot from 2022-09-07 18-10-39

### What issues does this PR fix or reference?

Fixes https://github.com/eclipse/che/issues/21635, and potentially https://github.com/eclipse/che/issues/21639
Could solve  as well

### How to test this PR?

1. Deploy Che with
```
chectl server:deploy -p=minikube -i=quay.io/vgulyy/che-server:test-fetching-git-files
```

2. Create a workspace with a factory
https://github.com/che-samples/java-spring-petclinic/tree/devfilev2?che-editor=eclipse/che-theia/next

3. Open Plugins view, check for installed plugins. Open terminal in editor container, check for `.vsix` files existence in `/plugins/sidecars/tools` directory.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by svor — APPROVED** [review]
LGTM
screenshot-eclipse-che apps ci-ln-zbsx2bb-72292 origin-ci-int-gce dev rhcloud com-2022 09 08-09_49_21
screenshot-eclipse-che apps ci-ln-zbsx2bb-72292 origin-ci-int-gce dev rhcloud com-2022 09 08-09_42_50

**Comment by nickboldt** [comment]
This needs to be backported to 7.52.x branch for DS 3.2.

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java [hunk 1]
```diff
@@ -139,14 +139,16 @@ protected boolean isPublicRepository(T remoteFactoryUrl) {
...
+  protected String formatUrl(String fileURL) throws DevfileException {
     String requestURL;
     try {
       if (new URI(fileURL).isAbsolute()) {
         requestURL = fileURL;
       } else {
-        // since files retrieved via REST, we cannot use path symbols like . ./ so cut them off
-        requestURL = remoteFactoryUrl.rawFileLocation(fileURL.replaceAll("^[/.]+", ""));
+        // since files retrieved via REST, we cannot use path like '.' or one that starts with './'
+        // or '../', so cut them off
```
Comment (tolusha) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java:149]: If we cut `../` of then we will have a wrong location, right?

Comment (vitaliy-guliy) [reply wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/scm/AuthorizingFileContentProvider.java:149]: We addressing the resource, beginning from the repository root, and this cutting does not change anything.
But yes, it looks a bit confusing, just take a look at the test https://github.com/eclipse-che/che-server/blob/main/wsmaster/che-core-api-factory-bitbucket-server/src/test/java/org/eclipse/che/api/factory/server/bitbucket/BitbucketServerAuthorizingFileContentProviderTest.java#L115

Let me try to remove cutting of `../`, update the PR and republish `che-server` image, that is used to test.

**Pull request merged**